c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -2837,14 +2837,38 @@ void local_raw_delete(void* ptr);
 
 class NativeCachingAllocator : public CUDAAllocator {
  private:
-  std::mutex mutex;
+  static constexpr size_t kNumMutexShard = 1031;
+
+  // TODO: change to hardware_destructive_interference_size when C++17 is available.
```

Comment:
PyTorch is fully c++17 compatible. Feel free to use it!

Reply:
thanks! It's actually not available in gcc/clang let me update the comment.